bcda/utils/common.go

```diff
@@ -79,3 +79,25 @@ func CloseFileAndLogError(f *os.File) {
 		logrus.Error(ferr)
 	}
 }
+
+// This is function that takes a string slice and removes any duplicates
+func Dedup(slice []string) []string {
+	// Get the length of the slice
+	var n = len(slice)
+
+	// Use the length to allocate memory once for new slice and map
+	var newslice = make([]string, 0, n)
+	var dupcheck = make(map[string]bool, n)
```

Comment:
If you'd want to save the most space, you can use a `map[string]struct{}`, since you're just looking for existence. The empty struct doesn't occupy any memory.

Usage would be:
```
if _, found := dupcheck[v]; !found {
    dupcheck[v] = struct{}{}
}
```

Reply:
If you change it to using an empty struct I would document that heavily. Using a bool is much more explicit to checking existence!

Reply:
I think I'll keep it as a bool to be explicit. Bool takes up 1 byte, and this is not taking into consideration any compiler optimizations.

Reply:
Makes sense! I shouldn't have referenced the size savings. More of awareness if you see this in other spots of the code/in other projects. I think that it's Go's way of implementing sets w/o introducing a new container (or generics 😄 )